circleguard/gui.py

```diff
@@ -286,71 +313,345 @@ def scroll_to_bottom(self):
+        self.name = name
+        self.id = id
+        self.text = QLabel(name + f" (id: {self.id})")
+        layout = QGridLayout()
+        layout.addWidget(self.text, 0, 0, 1, 7)
+        if not dragging:
+            self.delete_button = QPushButton(self)
+            self.delete_button.setIcon(QIcon(str(resource_path("./resources/delete.png"))))
+            self.delete_button.setMaximumWidth(30)
+            self.delete_button.clicked.connect(partial(lambda check_id: self.remove_signal.emit(check_id), self.id))
+            layout.addWidget(self.delete_button, 0, 7, 1, 1)
+        self.setLayout(layout)
+
+
+class LoadableW(QFrame):
+    """
+    A widget representing a circleguard.Loadable, which can be dragged onto
+    a DetectWidget. Keeps track of how many LoadableWidgets have been created
+    as a static ID attribute.
+
+    W standing for widget.
+    """
+    ID = 0
+    remove_loadable_signal = pyqtSignal(int) # id of loadable to remove
+
+    def __init__(self, name, required_input_widgets):
+        super().__init__()
+        LoadableW.ID += 1
+        self.name = name
+        self.required_input_widgets = required_input_widgets
+
+        self.layout = QGridLayout()
+        title = QLabel(self)
+        self.loadable_id = LoadableW.ID
+        t = "\t" # https://stackoverflow.com/a/44780467/12164878
+        # double tabs on short names to align with longer ones
+        title.setText(f"{name}{t+t if len(name) < 5 else t}(id: {self.loadable_id})")
+
+        self.delete_button = QPushButton(self)
+        self.delete_button.setIcon(QIcon(str(resource_path("./resources/delete.png"))))
+        self.delete_button.setMaximumWidth(30)
+        self.delete_button.clicked.connect(partial(lambda loadable_id: self.remove_loadable_signal.emit(loadable_id), self.loadable_id))
+        self.layout.addWidget(title, 0, 0, 1, 7)
+        self.layout.addWidget(self.delete_button, 0, 7, 1, 1)
+        self.setLayout(self.layout)
+
+    # Resources for drag and drop operations:
+    # qt tutorial           https://doc.qt.io/qt-5/qtwidgets-draganddrop-draggableicons-example.html
+    # qdrag docs            https://doc.qt.io/qt-5/qdrag.html
+    # real example code     https://lists.qt-project.org/pipermail/qt-interest-old/2011-June/034531.html
+    # bad example code      https://stackoverflow.com/q/7737913/12164878
+    def mouseMoveEvent(self, event):
+        # 1=all the way to the right/down, 0=all the way to the left/up
+        x_ratio = event.pos().x() / self.width()
+        y_ratio = event.pos().y() / self.height()
+        self.drag = QDrag(self)
+        # https://stackoverflow.com/a/53538805/12164878
+        pixmap = DragWidget(self.name, self.loadable_id).grab()
+        # put cursor in the same relative position on the dragwidget as
+        # it clicked on the real Loadable widget.
+        self.drag.setHotSpot(QPoint(pixmap.width() * x_ratio, pixmap.height() * y_ratio))
+        self.drag.setPixmap(pixmap)
```

Comment:
weird concept. It's fine but just putting the cursor in the bottom right would be fine

Reply:
Meh, this behavior appears pretty much everywhere (dragging icons on the mac doc, or tabs on chrome). The only difference here is we're using a drag widget of a different size than the original.